src/components/ListView.js

```diff
@@ -1,17 +1,36 @@
-import React from "react";
-import { useCollection } from "react-firebase-hooks/firestore";
-import { db } from "../lib/firebase";
+import React, { Fragment } from 'react';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { db } from '../lib/firebase';
+import { firestore } from 'firebase';
 
 const ListView = () => {
   const [value, loading, error] = useCollection(
-    db.collection(localStorage.getItem("token")),
+    db.collection(localStorage.getItem('token')),
     {
       snapshotListenOptions: { includeMetadataChanges: true },
     },
   );
 
   const deleteItemHandler = (id) => {
-    db.collection(localStorage.getItem("token")).doc(id).delete();
+    db.collection(localStorage.getItem('token')).doc(id).delete();
+  };
+
+  const onPurchaseChange = (e) => {
+    if (e.target.checked) {
+      db.collection(localStorage.getItem('token'))
+        .doc(e.target.value)
+        .update({ lastPurchasedDate: new Date() });
+    }
+  };
+
+  const checkIfItemHasBeenPurchased = (lastPurchaseTimestamp) => {
+    const now = Date.now();
+    const oneDay = 24 * 60 * 60 * 1000;
+    let isMoreThanADay = false;
+    if (lastPurchaseTimestamp) {
+      isMoreThanADay = now - lastPurchaseTimestamp.toMillis() < oneDay;
+    }
+    return isMoreThanADay;
```

Comment:
This is really nice! Super descriptive and handy logic. Feels pretty bulletproof. Nice work!

Reply:
Agreed! Love the variable naming going on here!